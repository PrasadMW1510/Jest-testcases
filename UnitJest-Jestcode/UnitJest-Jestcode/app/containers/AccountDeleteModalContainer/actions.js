/*
 *
 * AccountDeleteModalContainer actions
 *
 */
import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteRequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteMIARequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_MIA_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete is a success
 *
 *  @returns {{type}}
 */
export function postAccountDeleteRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account delete is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountDeleteRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_FAILURE,
    error,
  });
}

/**
 * Handles API logic for account unenroll action
 *
 * @param accountUnenrollPayload
 * @returns {{type}}
 */
export function postAccountUnenrollRequest(accountUnenrollPayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST,
    payload: accountUnenrollPayload,
  };
}

/**
 * Handles API logic for account unenroll is a success
 *
 *  @returns {{type}}
 */
export function postAccountUnenrollRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account unenroll is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountUnenrollRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_FAILURE,
    error,
  });
}
