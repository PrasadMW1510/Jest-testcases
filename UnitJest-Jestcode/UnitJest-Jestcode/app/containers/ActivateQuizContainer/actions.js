/*
 *
 * ActivateQuizContainer actions
 *
 */

import * as Constants from './constants';

export function activateQuizRequest(activateQuizPreviewData, searchOpts, searchDataChecked) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST,
    activateQuizPreviewData,
    searchOpts,
    searchDataChecked,
  };
}

export function getPostActivateRequestFailure(error) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST_FAILURE,
    error,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}

export function initialiseModalFlag() {
  return {
    type: Constants.INITIALIZE_MODAL_FLAG,
  };
}
