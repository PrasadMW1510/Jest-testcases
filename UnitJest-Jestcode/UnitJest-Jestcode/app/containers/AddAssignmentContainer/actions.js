/*
 *
 * AddAssignmentContainer actions
 *
 */

import * as Constants from './constants';

export function getStudentDetails(data) {
  return {
    type: Constants.GET_STUDENT_DETAILS,
    data,
  };
}
export function getStudentDetailsSuccess(data) {
  return {
    type: Constants.GET_STUDENTDETAIL_SUCCESS,
    data,
  };
}

export function getStudentDetailsFailure(error) {
  return {
    type: Constants.GET_STUDENTDETAIL_FAILURE,
    error,
  };
}

export function postSaveNewAssignment(data) {
  return {
    type: Constants.POST_NEW_ASSIGNMENT,
    data,
  };
}

export function postSaveNewAssignmentSuccess(data) {
  return {
    type: Constants.POST_NEW_ASSIGNMENT_SUCCESS,
    data,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}
