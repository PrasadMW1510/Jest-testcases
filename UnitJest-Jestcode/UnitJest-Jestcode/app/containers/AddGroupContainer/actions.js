/*
 *
 * AddGroupContainer actions
 *
 */

import * as Constants from './constants';

export function getClassesWithStudentInfo() {
  return {
    type: Constants.GET_CLASSES_WITH_STUDENT_INFO,
  };
}

export function getClassesWithStudentInfoSuccess(classInfo) {
  return {
    type: Constants.GET_CLASSES_WITH_STUDENT_INFO_SUCCESS,
    classInfo,
  };
}

export function getClassesWithStudentInfoFailure(error) {
  return {
    type: Constants.GET_CLASSES_WITH_STUDENT_INFO_FAILURE,
    error,
  };
}

export function postGroup(group) {
  return {
    type: Constants.POST_GROUP,
    group,
  };
}

export function postGroupSuccess(groupInfo) {
  return {
    type: Constants.POST_GROUP_SUCCESS,
    groupInfo,
  };
}

export function postGroupFailure(error) {
  return {
    type: Constants.POST_GROUP_FAILURE,
    error,
  };
}

export function resetGroupStatus() {
  return {
    type: Constants.RESET_GROUP_STATUS,
  };
}

export function getGroupInfo() {
  return {
    type: Constants.GET_GROUP_INFO,
  };
}

export function getGroupInfoSuccess(groupDetails) {
  return {
    type: Constants.GET_GROUP_INFO_SUCCESS,
    groupDetails,
  };
}

export function getGroupInfoFailure(error) {
  return {
    type: Constants.GET_GROUP_INFO_FAILURE,
    error,
  };
}
