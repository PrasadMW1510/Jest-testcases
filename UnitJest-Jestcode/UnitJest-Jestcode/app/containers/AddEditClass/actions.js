/*
 *
 * AddEditClass actions
 *
 */

import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API Logic for saving class data
 *
 * @param classObject
 * @returns {{type, classObject: *}}
 */
export function saveClassRequest(classObject, data = {}) {
  return {
    type: Constants.SAVE_CLASS_REQUEST,
    classObject,
    isEdit: !!data.edit,
  };
}

/**
 * Handles API logic for when saving class data is successful
 *
 * @returns {{type}}
 */
export function saveClassRequestSuccess() {
  return {
    type: Constants.SAVE_CLASS_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when saving class data is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function saveClassRequestFailure(error) {
  return {
    type: Constants.SAVE_CLASS_REQUEST_FAILURE,
    error,
  };
}

/**
 * Handles API Logic for saving class data
 *
 * @param classObject
 * @returns {{type, classObject: *}}
 */
export function saveClassMIARequest(classObject, data = {}) {
  return {
    type: Constants.SAVE_CLASS_MIA_REQUEST,
    classObject,
    isEdit: !!data.edit,
    editClassId: data.editClassId,
    schoolIdForClass: data.schoolIdForClass,
    searchOpts: data.searchOpts,
  };
}

/**
 * Handles API logic for when saving class data is successful
 *
 * @returns {{type}}
 */
export function saveClassMIARequestSuccess() {
  return {
    type: Constants.SAVE_CLASS_MIA_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when saving class data is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function saveClassMIARequestFailure(error) {
  return {
    type: Constants.SAVE_CLASS_MIA_REQUEST_FAILURE,
    error,
  };
}

/**
 * Handles API Logic for initializing class form
 *
 * @param id
 * @returns {{type, classObject: *}}
 */
export function initializeClassFormRequest(data) {
  return {
    type: Constants.INITIALIZE_CLASS_FORM_REQUEST,
    data,
  };
}

/**
 * Handles API logic for initializing class form success
 *
 * @returns {{type}}
 */
export function initializeClassFormRequestSuccess() {
  return {
    type: Constants.INITIALIZE_CLASS_FORM_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for initializing class form failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function initializeClassFormRequestFailure(error) {
  return generalFailure({
    type: Constants.INITIALIZE_CLASS_FORM_REQUEST_FAILURE,
    error,
  });
}
