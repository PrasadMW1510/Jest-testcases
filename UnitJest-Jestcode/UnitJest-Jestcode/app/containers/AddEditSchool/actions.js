/*
 *
 * AddSchoolContainer actions
 *
 */

import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Dispatched when the "Add a School" form is first opened
 * to retrieve whatever meta data is needed to populate it.
 */
export function getMetaDataRequest() {
  return {
    type: Constants.GET_META_DATA_REQUEST,
  };
}

/**
 * Dispatched when the retrieval of all meta data for the
 * "Add a School" form is complete (and successful).
 */
export function getMetaDataRequestSuccess() {
  return {
    type: Constants.GET_META_DATA_REQUEST_SUCCESS,
  };
}

/**
 * Dispatched when there's a failure during the retrieval
 * of meta data for the "Add a School" form.
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function getMetaDataRequestFailure(error) {
  return {
    type: Constants.GET_META_DATA_REQUEST_FAILURE,
    error,
  };
}

/**
 * Dispatched when the list of grades for the whole district is available.
 *
 * @param grades
 * @returns {{type, grades: *}}
 */
export function gradeListForDistrict(gradeList = []) {
  return {
    type: Constants.GRADE_LIST_FOR_DISTRICT,
    gradeList,
  };
}

/**
 * Handles API Logic for initializing school form
 *
 * @param id
 * @returns {{type, schoolObject: *}}
 */
export function initializeSchoolFormRequest(data) {
  return {
    type: Constants.INITIALIZE_SCHOOL_FORM_REQUEST,
    data,
  };
}

/**
 * Handles API logic for initializing school form success
 *
 * @returns {{type}}
 */
export function initializeSchoolFormRequestSuccess() {
  return {
    type: Constants.INITIALIZE_SCHOOL_FORM_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for initializing school form failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function initializeSchoolFormRequestFailure(error) {
  return generalFailure({
    type: Constants.INITIALIZE_SCHOOL_FORM_REQUEST_FAILURE,
    error,
  });
}

/**
 * Handles API logic for adding school
 *
 * @param schoolObject
 * @returns {{type, schoolObject: *}}
 */
export function saveSchoolRequest(schoolObject, data = {}) {
  return {
    type: Constants.SAVE_SCHOOL_REQUEST,
    schoolObject,
    isEdit: !!data.edit,
  };
}

/**
 * Handles API logic for adding MIA school
 *
 * @param schoolObject
 * @returns {{type, schoolObject: *}}
 */
export function saveSchoolMIARequest(schoolObject, data = {}) {
  return {
    type: Constants.SAVE_SCHOOL_MIA_REQUEST,
    schoolObject,
    isEdit: !!data.edit,
    editSchoolId: data.editSchoolId,
    searchOpts: data.searchOpts,
  };
}

/**
 * Handles API logic for when adding school is successful
 *
 * @returns {{type}}
 */
export function saveSchoolRequestSuccess() {
  return {
    type: Constants.SAVE_SCHOOL_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when adding school is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function saveSchoolRequestFailure(error) {
  return {
    type: Constants.SAVE_SCHOOL_REQUEST_FAILURE,
    error,
  };
}
