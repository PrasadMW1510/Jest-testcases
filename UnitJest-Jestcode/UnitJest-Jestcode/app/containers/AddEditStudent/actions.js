/*
 *
 * AddEditStudent actions
 *
 */

import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API Logic for saving student data
 *
 * @param studentObj
 * @returns {{type, studentObj: *}}
 */
export function saveStudentRequest(studentObj, data = {}) {
  return {
    type: Constants.SAVE_STUDENT_REQUEST,
    studentObj,
    isEdit: !!data.edit,
  };
}

/**
 * Handles API logic for when saving student data is successful
 *
 * @returns {{type}}
 */
export function saveStudentRequestSuccess() {
  return {
    type: Constants.SAVE_STUDENT_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when saving student data is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function saveStudentRequestFailure(error) {
  return {
    type: Constants.SAVE_STUDENT_REQUEST_FAILURE,
    error,
  };
}

/**
 * Handles API Logic for initializing student form
 *
 * @param data
 * @returns {{type, studentObject: *}}
 */
export function initializeStudentFormRequest(data) {
  return {
    type: Constants.INITIALIZE_STUDENT_FORM_REQUEST,
    data,
  };
}

/**
 * Handles API logic for initializing student form success
 *
 * @returns {{type}}
 */
export function initializeStudentFormRequestSuccess() {
  return {
    type: Constants.INITIALIZE_STUDENT_FORM_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for initializing student form failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function initializeStudentFormRequestFailure(error) {
  return generalFailure({
    type: Constants.INITIALIZE_STUDENT_FORM_REQUEST_FAILURE,
    error,
  });
}
