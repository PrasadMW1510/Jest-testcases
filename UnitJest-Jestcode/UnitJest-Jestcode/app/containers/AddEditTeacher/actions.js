/*
 *
 * AddEditTeacher actions
 *
 */

import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API logic for making calls needed for
 * Add Edit Teacher Modal
 *
 * @returns {{type}}
 */
export function addEditTeacherRequest(editTeacherId) {
  return {
    editTeacherId,
    type: Constants.ADD_EDIT_TEACHER_REQUEST,
  };
}

/**
 * Handles API logic for making calls needed for
 * Add Edit Teacher Modal are successful
 *
 * @returns {{type}}
 */
export function addEditTeacherRequestSuccess() {
  return {
    type: Constants.ADD_EDIT_TEACHER_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for making calls needed for
 * Add Edit Teacher Modal are a failure
 *
 * @param error
 * @returns {*}
 */
export function addEditTeacherRequestFailure(error) {
  return generalFailure({
    type: Constants.ADD_EDIT_TEACHER_REQUEST_FAILURE,
    error,
  });
}

/**
 * handles API logic for adding a new teacher
 *
 * @param profileData
 * @param permissionIds
 * @returns {{type, profileData, permissionIds}}
 */
export function postAddTeacherRequest(profileData = {}, permissionIds = []) {
  return {
    type: Constants.POST_ADD_TEACHER_REQUEST,
    profileData,
    permissionIds,
  };
}

/**
 * handles API logic for adding a new teacher are successful
 *
 * @returns {{type}}
 */
export function postAddTeacherRequestSuccess() {
  return {
    type: Constants.POST_ADD_TEACHER_REQUEST_SUCCESS,
  };
}

/**
 * handles API logic for adding a new teacher failure case
 *
 * @returns {{type}}
 */
export function postAddTeacherRequestFailure() {
  return {
    type: Constants.POST_ADD_TEACHER_REQUEST_FAILURE,
  };
}

/**
 * Handles API logic for updating profile data
 *
 * @param profileData
 * @returns {{type, profileData: {}}}
 */
export function postSaveTeacherRequest(profileData = {}, permissionIds = []) {
  return {
    type: Constants.POST_SAVE_TEACHER_REQUEST,
    profileData,
    permissionIds,
  };
}

/**
 * Handles API logic for updating profile data are successful
 *
 * @returns {{type}}
 */
export function postSaveTeacherRequestSuccess() {
  return {
    type: Constants.POST_SAVE_TEACHER_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for updating profile data are a failure
 *
 * @returns {{type}}
 */
export function postSaveTeacherRequestFailure() {
  return {
    type: Constants.POST_SAVE_TEACHER_REQUEST_FAILURE,
  };
}

/**
 * Handles API logic for saving teacher MIA
 *
 * @param profileData
 * @returns {{type, profileData: {}}}
 */
export function postSaveTeacherMIARequest(profileData = {}, permissionIds = [], data = {}) {
  return {
    type: Constants.POST_SAVE_TEACHER_MIA_REQUEST,
    profileData,
    permissionIds,
    editTeacherId: data.editTeacherId,
    searchOpts: data.searchOpts,
  };
}

/**
 * Dispatched when the profile details were retrieved successfully
 * is complete (and successful).
 */
export function addEditTeacherRequestProfileDetailsSuccess(profileDetails) {
  return {
    profileDetails,
    type: Constants.ADD_EDIT_TEACHER_REQUEST_PROFILE_DETAILS_SUCCESS,
  };
}

/**
 * Dispatched when the permissions details were retrieved successfully
 * is complete (and successful).
 */
export function addEditTeacherRequestPermissionsSuccess(permissionsData) {
  return {
    permissionsData,
    type: Constants.ADD_EDIT_TEACHER_REQUEST_PERMISSIONS_SUCCESS,
  };
}

/**
 * Dispatched when the permissions details were retrieved successfully
 * is complete (and successful).
 */
export function addEditTeacherRequestSchoolsAndClassesSuccess(schoolsAndClassesDetails) {
  return {
    schoolsAndClassesDetails,
    type: Constants.ADD_EDIT_TEACHER_REQUEST_SCHOOL_AND_CLASSES_SUCCESS,
  };
}

/**
 * Dispatched when the permissions details were retrieved successfully
 * is complete (and successful).
 */
export function addEditTeacherRequestPasswordConfigSuccess(passwordConfig) {
  return {
    passwordConfig,
    type: Constants.ADD_EDIT_TEACHER_REQUEST_PASSWORD_CONFIG_SUCCESS,
  };
}
