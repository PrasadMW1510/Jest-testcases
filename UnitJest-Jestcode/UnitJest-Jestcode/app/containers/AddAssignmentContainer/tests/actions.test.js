import * as Actions from '../actions';

describe('Print Book Label Container Action', () => {
  it('should return the correct constant for getStudentDetails', () => {
    expect(Actions.getStudentDetails()).toMatchSnapshot();
  });
  it('should return the correct constant for getStudentDetailsSuccess', () => {
    expect(Actions.getStudentDetailsSuccess()).toMatchSnapshot();
  });
  it('should return the correct constant for postSaveNewAssignment', () => {
    expect(Actions.postSaveNewAssignment()).toMatchSnapshot();
  });
  it('should return the correct constant for postSaveNewAssignmentSuccess', () => {
    expect(Actions.postSaveNewAssignmentSuccess()).toMatchSnapshot();
  });
  it('should return the correct constant for getStudentDetailsFailure', () => {
    expect(Actions.getStudentDetailsFailure()).toMatchSnapshot();
  });
  it('should return the correct constant for defaultAction', () => {
    expect(Actions.defaultAction()).toMatchSnapshot();
  });
});
