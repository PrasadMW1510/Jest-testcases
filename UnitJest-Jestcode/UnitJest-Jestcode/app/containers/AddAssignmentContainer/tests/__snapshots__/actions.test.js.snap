// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Print Book Label Container Action should return the correct constant for defaultAction 1`] = `
Object {
  "type": "app/AddAssignmentContainer/DEFAULT_ACTION",
}
`;

exports[`Print Book Label Container Action should return the correct constant for getStudentDetails 1`] = `
Object {
  "data": undefined,
  "type": "app/AddAssignmentContainer/GET_STUDENT_DETAILS",
}
`;

exports[`Print Book Label Container Action should return the correct constant for getStudentDetailsFailure 1`] = `
Object {
  "error": undefined,
  "type": "app/AddAssignmentContainer/GET_STUDENTDETAIL_FAILURE",
}
`;

exports[`Print Book Label Container Action should return the correct constant for getStudentDetailsSuccess 1`] = `
Object {
  "data": undefined,
  "type": "app/AddAssignmentContainer/GET_STUDENTDETAIL_SUCCESS",
}
`;

exports[`Print Book Label Container Action should return the correct constant for postSaveNewAssignment 1`] = `
Object {
  "data": undefined,
  "type": "app/AddAssignmentContainer/POST_NEW_ASSIGNMENT",
}
`;

exports[`Print Book Label Container Action should return the correct constant for postSaveNewAssignmentSuccess 1`] = `
Object {
  "data": undefined,
  "type": "app/AddAssignmentContainer/POST_NEW_ASSIGNMENT_SUCCESS",
}
`;
